functions/go/set-image/image_transformer.go

```diff
@@ -39,6 +43,53 @@ type SetImage struct {
 	Image types.Image `json:"image,omitempty" yaml:"image,omitempty"`
 	// AdditionalImageFields is used to specify additional fields to set image.
 	AdditionalImageFields types.FsSlice `json:"additionalImageFields,omitempty" yaml:"additionalImageFields,omitempty"`
+	// setImageResults is used internally to track which images were updated
+	setImageResults setImageResults
+}
+
+// setImageResultKey is used as a unique identifier for set image results
+type setImageResultKey struct {
+	ResourceRef yaml.ResourceIdentifier
+	// FilePath is the file path of the resource
+	FilePath string
+	// FileIndex is the file index of the resource
+	FileIndex int
+	// FieldPath is field path of the image field
+	FieldPath string
+}
+
+// setImageResult maps a previous image value to a new image value where set-image is applied
+// e.g. "nginx:1.20.2" -> "nginx:1.21.6"
+type setImageResult map[string]string
+
+// setImageResults tracks the number of images updated matching the key
+type setImageResults map[setImageResultKey]setImageResult
+
+// SdkResults returns sdk.Results representing which images were updated
+func (si *SetImage) SdkResults() sdk.Results {
+	var results sdk.Results
+	if len(si.setImageResults) == 0 {
+		results = append(results, &sdk.Result{
+			Message:  "no images changed",
+			Severity: sdk.Info,
+		})
+		return results
+	}
+	for k, v := range si.setImageResults {
+		imageJson, err := json.Marshal(v)
+		if err != nil {
+			return nil
+		}
+		resourceRef := k.ResourceRef
+		results = append(results, &sdk.Result{
+			Message:     fmt.Sprintf("set image: %s", imageJson),
```

Comment:
imageJson is something like `{old-val: new-val}`. If we always only have one key-value pair, why do we use it in json format. Can we do something like `set image from %q to %q`?

Reply:
It does not always have one key-value pair, see this unit test:
https://github.com/GoogleContainerTools/kpt-functions-catalog/pull/749/files#diff-b85391f699436541ad10a25bf2aeb38cac0e0a1c0969482bc9d132777137fb51R348

If the FieldPath is to a list element (e.g. `spec.containers.image`), and multiple list elements have changed image fields, then there will be multiple key-value pairs

Reply:
This behavior can of course be changed if this does not seem intuitive, I am open to feedback

Reply:
If it changes 2 image fields in containers, I'd expect something like:
```
spec.containers[0].image: set image from foo:1.2.3 to bar:4.5.6
spec.containers[1].image: set image from foo:latest to bar:4.5.6
```
WDYT?

Reply:
I think this comes down to an implementation detail of kustomize again, since `FieldPath` in this case is tracked as something like `spec.containers.image`. I could see a case for making this change, although I don't believe there is a straightforward way to include the index of a `SequenceNode` in `FieldPath` without changing the behavior of how `FieldPath` is tracked in kustomize.

We could alternatively do something like this given the current API:
```
spec.containers.image: set image from foo:1.2.3 to bar:4.5.6
spec.containers.image: set image from foo:latest to bar:4.5.6
```

WDYT?

Reply:
This looks better. We can do this for now.

Reply:
Done